### ISSUE TITLE
[DatePicker] [A11y] Keyboard navigation doesn't work with "next month" button

### ISSUE THREAD

**Issue by mefsap** [material-components/material-components-android#3387]
**Description:** When using an attached Bluetooth keyboard to navigate through the Date Picker, pressing either "tab" or "down arrow" on the "next month" button actually moves to the _previous_ month and then navigates to some date inside the calendar (I couldn't detect a pattern). 

**Expected behavior:** I would expect the tab key to navigate to the first date inside the calendar in this case. 

**Source code:** The issue can be reproduced with the M3 Catalog app (see attached video). 

**Minimal sample app repro:** The issue can be reproduced with the M3 Catalog app (see attached video). 

**Android API version:** Upside Down Cake

**Material Library version:** 1.9.0+ (reproduced building the M3 catalog from master, but also appears to exist on 1.9.0) 

**Device:** Pixel 5

https://github.com/material-components/material-components-android/assets/26826628/293a23e5-eb50-441f-ae9c-0db4ed88224a
